### ISSUE TITLE
VB.NET - Incorrect error BC30506 when compiling with .NET 5.0

### ISSUE THREAD

**Issue by ehasis** [dotnet/roslyn#49470]
Migrating a solution from .NET Core 3.1 to .NET 5.0 fail to compile. I was able to reproduce the problem with a minimal project as described on the steps bellow.

**Version Used**: 16.8.0

**Steps to Reproduce**:
1. Create a VB .NET Core Console Application (ConsoleApp1)
2. Add a new VB .NET Library project (Core) (ClassLibrary1)
3. Change TFM to .NET 5.0 on both projects
4. Make the ConsoleApp1 project references the ClassLibrary1
5. Add the following classes on ClassLibrary1
```vb
Public Class Button
    Public Event Click As EventHandler
End Class

Public Class Base
    Protected WithEvents Button1 As Button
End Class
```
6. Add the following class on ConsoleApp1
```vb
Public Class Derived
    Inherits ClassLibrary1.Base
    Private Sub Button1_Click(sender As Object, e As EventArgs) Handles Button1.Click
    End Sub
End Class
```

**Expected Behavior**:
The solution successfully compile.

**Actual Behavior**:
Fail compile the line `Private Sub Button1_Click(sender As Object, e As EventArgs) Handles Button1.Click` with:
error BC30506: Handles clause requires a WithEvents variable defined in the containing type or one of its base types.

**Comment by jaredpar** [comment]
Yeah I misinterpreted the problem here. 

The underlying issue has to do with the reference assemblies being produced here. Are you using any custom tooling for reference assemblies or is this a pretty standard project (in which case it would be our reference assembly generation code)

**Comment by ehasis** [comment]
Is just a standard project, no custom tooling.

**Comment by jaredpar** [comment]
This looks like an issue with our reference assembly generation then.

**Comment by ehasis** [comment]
Just a information. A couple months ago, I tried a migration when was .NET 5.0-preview7 and VS 16.8.0-preview1 the project compiled fine. So, something happened more recently.

**Comment by jcouv** [comment]
Fix was merged into 15.9p5. Thanks
